### PR TITLE
[skpi ci] Add pre checks early in the playbook

### PR DESCRIPTION
**Pull request by leseb**
refact the differents check present in ceph-ansible to fail early in the playbook if something is wrong like a bad variable definition or a requirement not met.
That way the playbook will fail before the deployment has started.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1538330
Signed-off-by: Sébastien Han <seb@redhat.com>

### CODE REVIEW

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-purge_bluestore_osds_container

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-purge_bluestore_osds_container

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-purge_bluestore_osds_container

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-docker_cluster_collocation

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-purge_bluestore_osds_container

**Comment by leseb** [comment]
jenkins test luminous-ansible2.4-docker_cluster_collocation

File: roles/ceph-defaults/tasks/checks.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - not osd_auto_discovery|default(False)
+    - osd_scenario|default('dummy') != 'lvm'
+    - item.failed
+
+- name: test if dedicated_devices are devices
+  parted:
+    device: "{{ item }}"
+    unit: MiB
+  with_items:
+    - "{{ dedicated_devices }}"
```
Comment (leseb) [inline roles/ceph-defaults/tasks/checks.yml]: `dedicated_devices` is not always known when ceph-defaults runs, especially if running on collocated osd scenario.

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-centos7_cluster

**Comment by guits** [comment]
jenkins test luminous-ansible2.4-docker_cluster

**Comment by guits** [comment]
@leseb I guess we should close this PR since the validation module has been merged

**Pull request closed without merging**